cidc_api/models/templates/utils.py

```diff
@@ -0,0 +1,164 @@
+from collections import defaultdict
+from typing import List, OrderedDict, Set, Type
+
+from sqlalchemy.orm import Session
+
+from .file_metadata import Upload
+from .model_core import MetadataModel, with_default_session
+
+
+def _all_bases(cls: Type) -> Set[Type]:
+    return set(cls.__bases__).union([s for c in cls.__bases__ for s in _all_bases(c)])
+
+
+def _all_subclasses(cls: Type) -> Set[Type]:
+    return set(cls.__subclasses__()).union(
+        [s for c in cls.__subclasses__() for s in _all_subclasses(c)]
+    )
+
+
+def _get_global_insertion_order() -> List[MetadataModel]:
```

Comment:
Gonna defer this to @curlup or perhaps this was already reviewed as part of a previous commit?